src/animation/AnimationUtils.js

```diff
@@ -158,6 +158,50 @@ var AnimationUtils = {
 
 		}
 
+	},
+
+	clone: function ( source ) {
+
+		var sourceNodes = {};
+		var clonedNodes = {};
+
+		var clone = source.clone();
+
+		source.traverse( function ( node ) {
+
+			sourceNodes[ node.name ] = node;
+
+		} );
+
+		clone.traverse( function ( node ) {
+
+			clonedNodes[ node.name ] = node;
+
+		} );
+
+		clone.traverse( function ( node ) {
+
+			if ( ! node.isSkinnedMesh ) return;
+
+			var clonedMesh = node;
+			var sourceMesh = sourceNodes[ node.name ];
+			var sourceBones = sourceMesh.skeleton.bones;
+
+			clonedMesh.skeleton = sourceMesh.skeleton.clone();
+			clonedMesh.bindMatrix = sourceMesh.bindMatrix;
+
+			clonedMesh.skeleton.bones = sourceBones.map( function ( bone ) {
+
+				return clonedNodes[ bone.name ];
```

Comment:
Similarly do you assume bones are in a subtree?

Reply:
This assumes the bones are within the subtree of the object being cloned, but not necessarily descendants of the SkinnedMesh they're associated with.

Reply:
I think we need to check if bones are in a subtree and halt with warn/error log if they aren't. Because it isn't guaranteed, for example users might call `Utils.clone( object2 )` for the following graph. Sometimes even bones wouldn't be in the scene graph.

- object1
  - bone1
    - bone2
- object2
  - skinned mesh